### PR TITLE
show/hide images on the admin catalog/product page

### LINKED ISSUE #5993: admin/category_products_listing: Shouldn't the image-column be optional?
As indicated by a couple of up-votes in [this](https://github.com/zencart/zencart/pull/4222) PR that added the category/product images to the admin's Categories/Products listing, shouldn't the image column have a soft-setting to allow a quick way to disable that information from the view?

Not all sites want that additional 'clutter'.

### PR DESCRIPTION
**Pull request by proseLA**
fixes #5993

### CODE REVIEW

**Comment by scottcwilson** [comment]
Not sure a toggle is really needed; I thought this was just going to be another flag.

**Comment by lat9** [comment]
> Not sure a toggle is really needed; I thought this was just going to be another flag.

Per the @torvista comment in the associated issue, I believe that having a toggle (so that an admin has control of their environment) is a good thing.

**Comment by lat9** [comment]
@proseLA, what do you think the effort would be to make this a session-based setting for the admin.  That way, once they've chosen to see/not see the images the selection would be remembered for their current login.

**Comment by torvista** [comment]
This modification still downloads the images and but then offers the option to hide them.
So it saves screen space, offers a more compact listing, but does not have a speed advantage.

I think the most complete implementation of this would be to prevent images being downloaded and be session-maintained.
That's what I did.

**Comment by scottcwilson** [comment]
> > Not sure a toggle is really needed; I thought this was just going to be another flag.
> 
> Per the @torvista comment in the associated issue, I believe that having a toggle (so that an admin has control of their environment) is a good thing.

Literally nothing else in the project works this way, but Steve has yet another personal preference, and he has unlimited time and energy for complaining, so we'll do it his way?  

This doesn't seem like a great process.  But it's not important enough to me to fight over, so do what you think is best.

**Comment by torvista** [comment]
>he has unlimited time and energy for complaining
I'll bypass the fundamental rule of not replying when angry.

Scott, you have an unlimited ability to repeatedly insult people. You just don't learn.
Where is my complaint in this thread?
I don't complain, I contribute. Or rather I did.

**Comment by proseLA** [comment]
@torvista i am no longer a fan of page reloads.  i think modern website design is more about sending all of the data down to the browser, and then using css and javascript/jquery to style the data as needed for the page.  i think all of these image files should be reasonably small, and in addition, browser caching of image files means you are not continuously reloading these image files.  again, my opinion is that the speed savings by not downloading the image files are minimal (but i have not seen your image files), but feel free to prove me wrong.  i know in my experience, other shopping cart software's admin experience is far slower than zc.  

@scottcwilson i did not code this to steve's preference.  i coded it to what made sense to me.  in addition, i think the rationale that nothing else is done this way is not a good rationale.  either we want zc to move forward with new ideas, incorporating more modern ideas of web design, or we do not.  personally i am more a fan of the former.

File: admin/category_product_listing.php [hunk 1]
```diff
@@ -341,6 +347,12 @@
                     </div>
                   </td>
                 </tr>
+                <tr>
+                    <td></td>
+                    <td><input type="button" id="imageView" class="btn btn-info btn-xs" value="<?= $buttonText; ?>"/>
```
Comment (lat9) [inline admin/category_product_listing.php:352]: For an HTML5 input, there's no `/` at the end-of-tag.

File: admin/category_product_listing.php [hunk 2]
```diff
@@ -1293,7 +1305,24 @@
             jQuery("tr.product-listing-row td").not('.dataTableButtonCell').on('click', (function() {
                 window.location.href = productEditLink.replace('[*]', jQuery(this).parent().attr('data-pid'));
             })).css('cursor', 'pointer');
-        })
+        });
+        $(document).ready(function () {
+            $('#imageView').click(function () {
```
Comment (lat9) [inline admin/category_product_listing.php:1310]: Use `$('#imageView').on('click', function() {`

File: includes/classes/ajax/zcAjaxAdminSessionChange.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+        } elseif ($_SESSION[$_POST['name']] === true) {
+            $_SESSION[$_POST['name']] = false;
+            return $_POST['name'] . ' set to false!';
+        }
+
+        $_SESSION[$_POST['name']] = true;
+        return $_POST['name'] . ' set to true!';
+
+
```
Comment (lat9) [inline includes/classes/ajax/zcAjaxAdminSessionChange.php:20]: Remove unneeded empty lines, please.

**Review by lat9 — CHANGES_REQUESTED** [review]
A couple of nit-picks, as usual.

**Review by lat9 — APPROVED** [review]

**Pull request merged**